lib/puppet/type/file/content.rb

```diff
@@ -119,6 +120,16 @@ def insync?(is)
       result
     end
 
+    def property_matches?(current, desired)
+      return super unless [:mtime, :ctime].include?(checksum_type)
+      begin
+        DateTime.parse(current) >= DateTime.parse(desired)
+      rescue => detail
+        Puppet.debug "resource with checksum_type #{checksum_type} didn't contain a date in #{current} or #{desired}"
+        super
```

Comment:
I think we should explicitly check for `current` or `desired` being `:absent` or `nil` (whichever is the one that this method can be called with), and short-circuit the method. FWIW, `lib/puppet/type/file/mode.rb` only seems to handle `nil`, not `:absent`

I don't like calling into `DateTime.parse` and rescuing because there could be other reasons for failing to parse. Also, I think `DateTime.parse` exceptions should fail the resource.